crossdock/client/gauntlet/behavior.go

```diff
@@ -0,0 +1,485 @@
+					4: 4,
+				},
+			},
+		},
+		{
+			function: "TestMulti",
+			give: []interface{}{
+				bytep(100),
+				int32p(200),
+				int64p(300),
+				map[int16]string{1: "1", 2: "2", 3: "3"},
+				numberzp(gauntlet.NumberzEight),
+				useridp(42),
+			},
+			want: &gauntlet.Xtruct{
+				StringThing: stringp("Hello2"),
+				ByteThing:   bytep(100),
+				I32Thing:    int32p(200),
+				I64Thing:    int64p(300),
+			},
+		},
+		{
+			function: "TestMultiException",
+			details:  "Xception",
+			give:     []interface{}{stringp("Xception"), stringp("foo")},
+			wantError: &gauntlet.Xception{
+				ErrorCode: int32p(1001),
+				Message:   stringp("This is an Xception"),
+			},
+		},
+		{
+			function: "TestMultiException",
+			details:  "Xception2",
+			give:     []interface{}{stringp("Xception2"), stringp("foo")},
+			wantError: &gauntlet.Xception2{
+				ErrorCode:   int32p(2002),
+				StructThing: &gauntlet.Xtruct{StringThing: stringp("foo")},
+			},
+		},
+		{
+			function: "TestMultiException",
+			details:  "no error",
+			give:     []interface{}{stringp("hello"), stringp("foo")},
+			want:     &gauntlet.Xtruct{StringThing: stringp("foo")},
+		},
+		{
+			function: "TestNest",
+			give: []interface{}{
+				&gauntlet.Xtruct2{
+					ByteThing: bytep(-1),
+					I32Thing:  int32p(-1234),
+					StructThing: &gauntlet.Xtruct{
+						StringThing: stringp("0"),
+						ByteThing:   bytep(1),
+						I32Thing:    int32p(2),
+						I64Thing:    int64p(3),
+					},
+				},
+			},
+			want: &gauntlet.Xtruct2{
+				ByteThing: bytep(-1),
+				I32Thing:  int32p(-1234),
+				StructThing: &gauntlet.Xtruct{
+					StringThing: stringp("0"),
+					ByteThing:   bytep(1),
+					I32Thing:    int32p(2),
+					I64Thing:    int64p(3),
+				},
+			},
+		},
+		{
+			function: "TestSet",
+			give: []interface{}{
+				map[int32]struct{}{
+					1:  struct{}{},
+					2:  struct{}{},
+					-1: struct{}{},
+					-2: struct{}{},
+				},
+			},
+			want: map[int32]struct{}{
+				1:  struct{}{},
+				2:  struct{}{},
+				-1: struct{}{},
+				-2: struct{}{},
+			},
+		},
+		{
+			function: "TestString",
+			give:     []interface{}{stringp("hello")},
+			want:     "hello",
+		},
+		{
+			function: "TestStringMap",
+			give: []interface{}{
+				map[string]string{
+					"foo":   "bar",
+					"hello": "world",
+				},
+			},
+			want: map[string]string{
+				"foo":   "bar",
+				"hello": "world",
+			},
+		},
+		{
+			function: "TestStruct",
+			give: []interface{}{
+				&gauntlet.Xtruct{
+					StringThing: stringp("0"),
+					ByteThing:   bytep(1),
+					I32Thing:    int32p(2),
+					I64Thing:    int64p(3),
+				},
+			},
+			want: &gauntlet.Xtruct{
+				StringThing: stringp("0"),
+				ByteThing:   bytep(1),
+				I32Thing:    int32p(2),
+				I64Thing:    int64p(3),
+			},
+		},
+		{
+			function: "TestTypedef",
+			give:     []interface{}{useridp(42)},
+			want:     gauntlet.UserId(42),
+		},
+		{
+			function: "TestVoid",
+			give:     []interface{}{},
+		},
+	}
+
+	for _, tt := range tests {
```

Comment:
This block is massive, can you try and pull the main pieces out into separate funcs?